### PR TITLE
Domains: Link to the exact purchase when possible

### LINKED ISSUE #1187: Domains: Link domain payment settings to the correct manage purchase URL
The Payment Settings button in /domains/manage/:site/edit/:domain takes you to the list in /purchases instead of to the upgrade page (/purchases/:site/:upgradeId). To get the right purchase ID we'll need to load purchases data in this screen.

Noticed by @fabianapsimoes in https://github.com/Automattic/wp-calypso/pull/579.

### PR DESCRIPTION
**Pull request by klimeryk**
As mentioned in #1187, the `Payment Settings` link always leads to the full list of purchases. We should instead link to the exact purchase for the domain/redirect we're interested in.

A similar issue has been raised in #7464, only about the domain warnings we show for expiring/expired domains.

### Testing

Requires `D6123-code`.

screen shot 2017-06-25 at 01 39 24

Go to domain details on Domain Management. Verify that the `Payment Settings` link leads to the exact purchase details page (`/me/purchases/:siteId/:purchaseId`), not the whole list (`/me/purchases`).

screen shot 2017-06-25 at 18 01 23
screen shot 2017-06-25 at 18 02 02

For the domain warning for expiring or expired domains, it's a bit more tricky. They show up in three places - in the sidebar (as seen in the screenshots above), in the Plans page and in the Domain Management's domain list. Depending on the number of expiring/expired notices, we show a different copy:
* If it's just one domain, we should link to that exact domain's purchase.
* If it's more than one domain, we should link to `/me/purchases`, as previously.

Fixes #1187 and #7464

### CODE REVIEW

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 1]
```diff
@@ -614,17 +629,20 @@ export default React.createClass( {
...
+	render() {
 		debug( 'Domains:', this.getDomains() );
 		const notices = this.getPipe().map( renderer => renderer() ).filter( notice => notice );
-		return notices.length ? <div className="site__notices">{ notices }</div> : null;
+		return notices.length ? <div>{ notices }</div> : null;
 	}
-} );
+}
+
+export { DomainWarnings };
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Needed for tests :man_shrugging:

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 2]
```diff
@@ -614,17 +629,20 @@ export default React.createClass( {
...
 		}
-	},
+	}
 
-	render: function() {
+	render() {
 		debug( 'Domains:', this.getDomains() );
 		const notices = this.getPipe().map( renderer => renderer() ).filter( notice => notice );
-		return notices.length ? <div className="site__notices">{ notices }</div> : null;
+		return notices.length ? <div>{ notices }</div> : null;
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: The `site__notices` style is not defined anywhere anymore (plus, it was causing a linter warning).

**Comment by mikeshelton1503** [comment]
@klimeryk I was trying to test this out but when on the branch I get this error:

>Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object. Check the render method of `CurrentSite`.

When I try to view Domains.

**Comment by klimeryk** [comment]
@mikeshelton1503 Thanks for having a look! The error is probably because you need to have a sandbox and the `D6123-code` patch on it :( Sorry, I've mentioned it in the description, but I should have been more clear about that. I'll try to get it reviewed and merged ASAP, so that you can check out this frontend patch.

**Comment by hambai** [comment]
Hmm... I guess I get the same errors as @mikeshelton1503 even with patched and sandboxed public-api. Here's a screenshot of my console:
screen shot 2017-06-29 at 4 17 48 pm

**Comment by klimeryk** [comment]
Well, that's embarassing - sorry for doubting you, @mikeshelton1503 ;) (and thanks for having a look, @hambai!) I have no idea how that error got through, my bad 😞 I've fixed it now and thanks to @hambai's quick review, the backend changes have now been merged and deployed, so you can check this PR in all its glory.

**Comment by mikeshelton1503** [comment]
Looks good 👍  The back button is an issue. Confusing that clicking the back button takes you to a full list of purchases instead of back to your domains. However, that's not a new issue nor something we need to solve in this PR.

File: client/state/sites/domains/assembler.js [hunk 3]
```diff
@@ -33,7 +33,8 @@ export const createSiteDomainObject = domain => {
 		pointsToWpcom: Boolean( domain.points_to_wpcom ),
 		registrar: String( domain.registrar ),
 		registrationDate: String( domain.registration_date ),
+		subscriptionId: domain.subscription_id,
```
Comment (klimeryk) [inline client/state/sites/domains/assembler.js]: Oh, didn't know we had that for domains - will update, thanks!

**Pull request merged**